johannes_mueller/entry_user_interf2/Readme.md

```diff
@@ -0,0 +1,364 @@
+
+        FigureCanvas.__init__(self, self.fig)       # initialize canvas
+        FigureCanvas.setSizePolicy(self, QSizePolicy.Expanding,
+                                   QSizePolicy.Expanding)
+        FigureCanvas.updateGeometry(self)
+
+class matplotlibWidget(QWidget):
+    """
+    The matplotlibWidget class based on QWidget
+    """
+    def __init__(self, parent=None):
+        QWidget.__init__(self, parent)
+        # save canvas and toolbar
+        self.canvas = MplCanvas()
+        self.toolbar = NavigationToolbar(self.canvas, self)
+        # set layout and add them to widget
+        self.vbl = QVBoxLayout()
+        self.vbl.addWidget(self.toolbar)
+        self.vbl.addWidget(self.canvas)
+        self.setLayout(self.vbl)
+```
+
+Next, we have to import this widget definition into our MainWindow in the Qt Designer. To do so, add a blank widget (blue box) to the MainWindow and rename it ```plotWidget``` (see orange box):
+
+![designer_main_promote_1](https://user-images.githubusercontent.com/38459088/137171680-3230a5ee-1db9-4d86-b55d-c8043ea84667.png)
+
+Now, promote this widget to a matplotlib widget by ```Right-click widget -> Promote to...```. This way, we tell the designer that our currently useless widget will be promoted to a widget with some self-defined functionality. In the window that pops up, enter the following, and click ```Add -> Promote``` to promote the widget to a matplotlibwidget. Note that the entry in **Promoted class name** must refer to the name of the class we defined above in matplotlibwidgetFile.py ```class matplotlibWidget(QWidget):``` 
+
+![designer_main_promote_3](https://user-images.githubusercontent.com/38459088/137174317-a4be907d-9aca-4fd8-85f1-f96a7de3fdd0.JPG)
+
+
+The widget is now a matplotlib widget! Now, don't forget to save the .ui file and run ```pyuic5```. We can then set up the main window as before with the following code and display our new widget, that contains all the familiar tools of a matplotlib figure, such as the toolbar to pan/zoom/save images:
+
+```python
+import sys
+
+# This imports the previously generated UI file
+from MainWindow import Ui_MainWindow
+
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QApplication, QMainWindow
+
+# Define the the main window class
+class MainWindow(QMainWindow, Ui_MainWindow):
+    def __init__(self):
+        super().__init__()
+        
+        #Initialize GUI
+        self.setupUi(self)
+    
+    def closeEvent(self, event):
+        self.close()
+        app.quit()
+        
+
+# Start the application
+app = QApplication(sys.argv)
+
+window = MainWindow()
+window.show()
+
+app.exec_()
+
+```
+![simple_GUI2](https://user-images.githubusercontent.com/38459088/137174851-6ba7257f-62fd-40b4-9f74-603ac7524e5c.JPG)
+
+You can draw data on this figure similarly as you would usually do in matplotlib. If your widget in the designer is called ```plotWidget```, plotting a curve would look like this in the code:
+```python
+# Plot a curve:
```

Comment:
Suggestion: actually enter some array into x_data and y_data and show the result in a figure.
Maybe also mention this could be allocated inside the callback function of the button to refresh the graphic.
[Here](https://matplotlib.org/stable/gallery/widgets/buttons.html) is a reference from matplotlib, although not using Qt...
I think an animated gif of clicking the button and updating something in the graphic would be awesome! Not strictly necessary though.